doc/api/esm.md

```diff
@@ -681,126 +695,54 @@ of the following:
 | `'module'`   | Load an ES module              | { [`string`][], [`ArrayBuffer`][], [`TypedArray`][] }                      |
 | `'wasm'`     | Load a WebAssembly module      | { [`ArrayBuffer`][], [`TypedArray`][] }                                    |
 
+Note: When format is `'commonjs'`, the value of `source` is ignored.
```

Comment:
As a reader, this kind of doesn’t make sense. The whole purpose of `load` is to return the source code of a resource, and tell Node how to interpret it. If `source` is ignored, what is Node supposed to do with `{ format: 'commonjs' }`? Is that equivalent to `{ format: 'module', source: '' }`?

As a developer working on this feature, I understand that this is just an unimplemented feature that we haven’t gotten to. We might as well say so (or say that `format: 'commonjs'` will never be supported, if that’s the case).

Reply:
This note was at @guybedford's request.

>what is Node supposed to do with { format: 'commonjs' }? Is that equivalent to { format: 'module', source: '' }?

I think they are very much not equivalent: The former tells Node to switch to the CJS loader (and in the process, the value of `source` is dropped—I presume due to some logically necessity?). The latter tells Node to treat it as an ESM that has no content.

Sorry, I don't know the answer to whether it's merely unimplemented vs intentionally unsupported. I think Guy mentioned there was a _significant_ amount of work needed (something about aligning CJSLoader and ESMLoader's APIs?), so it basically just isn't gonna happen.